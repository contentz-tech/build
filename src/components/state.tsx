import { createContext, useContext } from "react";
import { jsx } from "@emotion/core";
import { IState } from "../state";

const StateContext = createContext<IState | null>(null);

interface StateProviderProps {
  state: IState;
  children: JSX.Element[] | JSX.Element;
}

function StateProvider({ state, children }: StateProviderProps) {
  return (
    <StateContext.Provider value={state}>{children}</StateContext.Provider>
  );
}

function useState(): IState {
  const state = useContext(StateContext);
  if (state === null)
    throw new Error(
      "Wrap your component in `StateProvider` and pass a valid state."
    );
  return state;
}

export { StateProvider, useState };
