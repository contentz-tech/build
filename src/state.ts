import { IConfig, config as getConfig } from "./getters/config";
import { articles, IArticles } from "./getters/articles";
import { pages, IPages } from "./getters/pages";
import { slides, ISlides } from "./getters/slides";
import { links, ILink } from "./getters/links";
import { statics } from "./getters/statics";
import { i18n } from "./getters/i18n";

interface IState {
  config: IConfig;
  articles: IArticles;
  pages: IPages;
  slides: ISlides;
  links: ILink[];
  statics: string[];
  i18n: any;
}

async function getState(): Promise<IState> {
  const config: IConfig = await getConfig();
  return {
    config,
    articles: await articles(config),
    pages: await pages(config),
    slides: await slides(config),
    links: await links(),
    statics: await statics(),
    i18n: await i18n(config)
  };
}

export { getState, IState };
