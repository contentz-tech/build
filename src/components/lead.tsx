import { jsx } from "@emotion/core";
import format from "date-fns/format";
import { useState } from "./state";

function Title(props: {
  children: JSX.Element[] | JSX.Element | string | undefined;
}) {
  return (
    <h1
      css={{
        fontSize: "2.25em",
        fontWeight: "bold",
        letterSpacing: "-0.028em",
        margin: "1em 0"
      }}
    >
      {props.children}
    </h1>
  );
}

function Description(props: {
  children: JSX.Element[] | JSX.Element | string | undefined;
}) {
  return (
    <p
      css={{
        borderLeft: "3px solid black",
        boxSizing: "border-box",
        paddingLeft: "calc(0.5em - 3px)",
        margin: "-1em 0 1em -0.5em",
        fontSize: "1.5em",
        fontWeight: 200
      }}
    >
      {props.children}
    </p>
  );
}

function Date(props: { date: Date }) {
  const state = useState();

  const locale = require(`date-fns/locale/${state.config.language}`);

  return (
    <time
      dateTime={props.date.toJSON()}
      css={{
        position: "absolute",
        right: "0.5em",
        bottom: "100%"
      }}
    >
      <strong>{format(props.date, "MMMM DD, YYYY", { locale })}</strong>
    </time>
  );
}

export { Title, Description, Date };
