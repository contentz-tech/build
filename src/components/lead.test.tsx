/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import { Title, Description, Date as DateTime } from "./lead";
import { StateProvider } from "./state";

const state = {
  config: { language: "en" },
  articles: {
    order: [],
    byPath: {}
  },
  slides: {
    order: [],
    byPath: {}
  },
  statics: [],
  pages: {
    order: [],
    byPath: {}
  },
  links: [],
  i18n: {
    patreon: {
      first: "Do you like my content?",
      link: "Become a Patreon and help me continue writing!"
    }
  }
}

describe("Lead", () => {
  test("Title", () => {
    const { asFragment, getByText } = render(<Title>Contentz.tech</Title>);
    
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Contentz.tech/)).toBeInTheDocument();
  });

  test("Description", () => {
    const { asFragment, getByText } = render(
      <Description>Create Content, Get a Highly Optimized Website</Description>
    );
    
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Create Content, Get a Highly Optimized Website/))
      .toBeInTheDocument();
  });

  test("Date", () => {
    const { asFragment, getByText } = render(
      <StateProvider state={state}>
        <DateTime date={new Date("2019-09-15T00:12:57.350Z")} />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/September 14, 2019/)).toBeInTheDocument();
  });
});
