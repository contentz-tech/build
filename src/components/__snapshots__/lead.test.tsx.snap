// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lead Date 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: absolute;
  right: 0.5em;
  bottom: 100%;
}

<time
    class="emotion-0"
    datetime="2019-09-15T00:12:57.350Z"
  >
    <strong>
      September 14, 2019
    </strong>
  </time>
</DocumentFragment>
`;

exports[`Lead Description 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-left: 3px solid black;
  box-sizing: border-box;
  padding-left: calc(0.5em - 3px);
  margin: -1em 0 1em -0.5em;
  font-size: 1.5em;
  font-weight: 200;
}

<p
    class="emotion-0"
  >
    Create Content, Get a Highly Optimized Website
  </p>
</DocumentFragment>
`;

exports[`Lead Title 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 2.25em;
  font-weight: bold;
  -webkit-letter-spacing: -0.028em;
  -moz-letter-spacing: -0.028em;
  -ms-letter-spacing: -0.028em;
  letter-spacing: -0.028em;
  margin: 1em 0;
}

<h1
    class="emotion-0"
  >
    Contentz.tech
  </h1>
</DocumentFragment>
`;
